devtools/x/src/cargo.rs

```diff
@@ -62,7 +62,7 @@ impl Cargo {
         // The environment is inherited for child processes so we only need to set RUSTUP_TOOLCHAIN
         // if it isn't already present in the environment
         if env::var_os(RUSTUP_TOOLCHAIN).is_none() {
-            inner.env(RUSTUP_TOOLCHAIN, RUST_TOOLCHAIN_VERSION);
+            inner.env(RUSTUP_TOOLCHAIN, RUST_TOOLCHAIN_VERSION.trim());
```

Comment:
This is a constant and should not need to be "trimmed" just update the constant to not have excess whitespace if needed

Reply:
Can't call trim in a constant block, at least that's what the compiler tells me.

Reply:
I'm not suggesting you call trim when creating the const, I'm suggesting you make sure the const has a value that doesn't need to be trimmed, which it currently doesn't.

Reply:
As per offline communication, since this is sourced from a file, this is fine to land.

Reply:
Yep, I didn't take the time I should have to actually look at how the constant itself was initialized. I never intended to block this PR from landing though (I would have requested changes if I had). I also suppose my initial question wasn't worded well enough to get at what I really wanted to know: What issue you were actually seeing without trimming and why trimming seemed to solve the issue.

Reply:
while doing some tests with sccache, I observed some versions of cargo (rustup wrapped cargo) fail when reading the environment variable.   Can't recall the exact setup now - but end result was a broken compilation.   Not entirely sure of all the impact that extra line could have had, as, when I saw that error the first time -- I just fixed and carried on fighting another bug in cargo.